tide/master/rest/RestController.cpp

```diff
@@ -0,0 +1,194 @@
+
+    server.handlePUT(_toggleSelectWindow);
+    connect ( &_toggleSelectWindow, &RestCommand::received,
+              this, &RestController::_handleToggleSelectWindow );
+
+    server.handlePUT(_unfocusWindow);
+    connect ( &_unfocusWindow, &RestCommand::received,
+              this, &RestController::_handleUnfocusWindow );
+
+    server.handlePUT(_moveWindowtoFullscreen);
+    connect ( &_moveWindowtoFullscreen, &RestCommand::received,
+              this, &RestController::_handleMoveWindowToFullscreen );
+
+    server.handlePUT( "tide/resizeWindow", [this]( const std::string& received )
+    { return _handleResizeWindow( received ); } );
+
+    server.handlePUT( "tide/moveWindow", [this]( const std::string& received )
+    { return _handleMoveWindow( received ); } );
+
+    server.handlePUT( "tide/focusWindows", [this]()
+    { return _handleFocusWindows(); });
+
+    server.handlePUT( "tide/unfocusWindows", [this]()
+    { return _handleUnfocusWindows(); });
+
+    server.handlePUT( "tide/exitFullScreen", [this]()
+    { return _handleExitFullScreen(); } );
+
+    server.handlePUT( "tide/deselectWindows", [this]()
+    { return _handleDeselectWindows(); } );
+
+    server.handlePUT( "tide/clear", [this]()
+    { _displayGroup.clear(); return true; } );
+}
+
+bool RestController::_handleMoveWindowToFront( const QString uri )
+{
+    _controller->moveWindowToFront( uri );
+    return true;
+}
+
+bool RestController::_handleFocusWindows()
+{
+    _controller->focusSelected();
+    return true;
+}
+bool RestController::_handleUnfocusWindow( const QString& id )
+{
+   _controller->unfocus( QUuid( id ) );
+    return true;
+}
+bool RestController::_handleUnfocusWindows()
+{
+    _controller->unfocusAll();
+    return true;
+}
+bool RestController::_handleMoveWindowToFullscreen( const QString uri )
+{
+    _controller->showFullscreen( uri );
+    return true;
+}
+bool RestController::_handleExitFullScreen()
+{
+    _controller->exitFullscreen();
+    return true;
+}
+
+bool RestController::_handleResizeWindow( const std::string& string )
+{
+    const auto obj = _toJSONObject( string );
+    if( obj.empty( ))
+        return false;
+
+    const auto windowSize = QSizeF( obj["w"].toString().toDouble(),
+                                    obj["h"].toString().toDouble( ));
+    auto window = _displayGroup.getContentWindow( obj["uri"].toString( ));
+    if( !window )
+       return false;
+    ContentWindowController controller( *window, _displayGroup );
+    if (obj["centered"].toString().toInt( ))
+        controller.resize( windowSize, WindowPoint::CENTER );
+    else
+        controller.resize( windowSize, WindowPoint::TOP_LEFT );
+    return true;
+}
+
+bool RestController::_handleMoveWindow( const std::string& string )
+{
+    const auto obj = _toJSONObject( string );
+    if( obj.empty( ))
+        return false;
+
+    const auto windowPos = QPointF( obj["x"].toString().toDouble(),
```

Comment:
move the windowPos declaration below the check for if( !window ), there is no need to evaluate it if there is no window (also in the previous function).